### ISSUE TITLE
No Visible @interface for 'TSBackgroundFetch'

### ISSUE THREAD

**Issue by masiamj** [transistorsoft/react-native-background-geolocation#459]
When upgrading our React Native iOS app, we're met with the error that there is 'No Visible @interface for 'TSBackgroundFetch' declares the selector 'performFetchWithCompletionHandler'.

## Environment
* Plugin version: 2.10.1
* Platform: iOS
* OS version: 11.0.x
* Device manufacturer / model: Apple iPhone 6s
* React Native version (`react-native -v`): 0.51.0
* Plugin config

```bash
        debug: Config.PRODUCTION_BUILD === 'no',
        desiredAccuracy: 0,
        distanceFilter: 10,
        foregroundService: true,
        headers,
        httpRootProperty: '.',
        locationTemplate: '{ "latitude":<%= latitude %>, "longitude":<%= longitude %> }',
        method: 'PUT',
        notificationColor: '#8BC53E',
        notificationIcon: 'drawable/ic_stat_bglocation',
        notificationTitle: 'Thumbster Connect',
        notificationText: 'Location Tracking Enabled',
        logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
        stationaryRadius: 25,
        startOnBoot: false,
        stopOnTerminate: false,
        url,
```

## Expected Behavior
App should build correctly on iOS.

## Actual Behavior
When building our iOS app, we're met with the error that there is 'No Visible @interface for 'TSBackgroundFetch' declares the selector 'performFetchWithCompletionHandler'. The app was building correctly previously with an older (0.44.x) version of React Native.

## Steps to Reproduce
1. `react-native run-ios`

## Context
Build the app with the Background Geolocation Plugin.

## Debug logs
![tsbackgroundfetch](https://user-images.githubusercontent.com/7084914/38329980-5a5f27bc-381d-11e8-850a-ac0b663a356b.png)

**Comment by christocracy** [comment]
You probably have an old version of background-fetch installed.

```
$ npm install —save react-Native-background-fetch
```

**Comment by christocracy** [comment]
Update please.

**Comment by masiamj** [comment]
Resolved. Problem was related to a React Native upgrade and the build step. Thanks for the advice.

**Comment by jmeyers91** [comment]
Just updated. Still seeing this issue with `react-native-background-fetch@2.4.1` in a fairly new React-Native project:

```
react-native-cli: 2.0.1
react-native: 0.55.3
```

**Comment by christocracy** [comment]
In `Build Phases -> Link Binary Wth Libraries`, verify following:

![](https://dl.dropboxusercontent.com/s/igkvt798oramv4h/Screenshot%202018-04-20%2015.01.34.png?dl=1)

**Comment by jmeyers91** [comment]
I was able to fix the issue by copying the up-to-date contents of [RNBackgroundFetch+AppDelegate.m](https://github.com/transistorsoft/react-native-background-fetch/blob/master/ios/RNBackgroundFetch/RNBackgroundFetch%2BAppDelegate.m) into my project. Updating didn't fix the issue because `react-native link` doesn't override files made through previous calls to `react-native link`. I might have been able to run `react-native unlink` first, but my experiences with unlinking native modules has been pretty poor, so I figured I'd do it manually.
